api/src/main/java/jakarta/data/restrict/Restrict.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (c) 2024 Contributors to the Eclipse Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package jakarta.data.restrict;
+
+import java.util.List;
+import java.util.Set;
+
+import jakarta.data.restrict.Restriction.Basic;
+import jakarta.data.restrict.Restriction.Operator;
+
+// TODO document
+// This is one of two places from which to obtain restrictions.
+// The other place is from static metamodel attributes.
+public enum Restrict {
```

Comment:
Not sure I love how this breaks the pattern established by `Sort`/`Order`.

Reply:
Is the enum class the only problem here? If so, I can move the enum to another class and make Restrict into a normal class like Order is.  I didn't include this in the commit with the first round of fixes because I want to be sure what you are asking for.

Reply:
Yes, I think that's my preference. I don't think it's going to be normal for the user to want to access the enum values directly.

Reply:
I agree. The user will typically never look the composite type. The enum values are mainly for the Jakarta Data provider to check which type of composite it is. I made Restrict into a normal class and moved the enum elsewhere. Thanks for pointing this out.